### ISSUE TITLE
Router is not applied to children Links

### ISSUE THREAD

**Issue by t73biz** [jpgorman/react-append-to-body#10]
I have tried to implement this to add in Nav elements to my navbar that is static. I receive an error message
```
Warning: Failed context type: The context `router` is marked as required in `Link`, but its value is `undefined`.
    in Link (created by App)
    in li (created by App)
    in ul (created by App)
    in AfterNav
    in span
```

The code I'm using is:

```
//Libraries
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Link, Switch } from 'react-router-dom';
import {componentWillAppendToBody} from 'react-append-to-body';

//Components
import Dashboard from './components/Dashboard';
import Chores from './components/Chores/Chores';

function AfterNav({children}) {
    return [children];
}

const AppendedAfterNav = componentWillAppendToBody(AfterNav);

class App extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            user: user
        }
    }

    render () {
        return (
                <Router>
                    <div>
                        <AppendedAfterNav subtreeContainer={"#var-nav"}>
                            <ul className="nav navbar-nav">
                                <li>
                                    <Link to="/chores">Chores</Link>
                                </li>
                                <li>
                                    <Link to="/dashboard">Dashboard</Link>
                                </li>
                            </ul>
                        </AppendedAfterNav>
                        <Switch>
                            <Route path="/dashboard" render={(routeProps) => (<Dashboard {...routeProps} user={this.state.user} />)}/>
                            <Route path="/chores" render={(routeProps) => (<Chores {...routeProps} user={this.state.user} />)}/>
                        </Switch>
                    </div>
                </Router>
        );
    };
}

ReactDOM.render(<App user={user}/>, document.getElementById('container'));

```

**Comment by jpgorman** [comment]
Thanks for feedback. Currently this module doesn't persist context to components that are rendered via `componentWillAppendToBody` so that Router context won't be pass into the Link components.

This relates back to issue #9 I'll make that change and publish and update. This would mean that you'd have to update to React/React DOM 16.x.x

**Comment by jpgorman** [comment]
I've released an updated version now 2.0.14 that resolves this issue by allowing context to be passed in. Also you won't need to update your version of React either :)

You can pass in the router context by adding the contextTypes to your appended component.

In your example it would be 

`
const AppendedAfterNav = componentWillAppendToBody(AfterNav);
AppendedAfterNav.contextTypes = {
  router: React.PropTypes.any.isRequired
};
`
